neo/Wallets/Wallet.cs

```diff
@@ -290,18 +291,49 @@ private Transaction MakeTransaction(Snapshot snapshot, TransactionAttribute[] at
                     ValidUntilBlock = snapshot.Height + Transaction.MaxValidUntilBlockIncrement,
                     Attributes = attributes
                 };
-                ContractParametersContext context = new ContractParametersContext(tx);
-                if (!Sign(context)) continue;
-                if (!context.Completed) continue;
-                tx.Witnesses = context.GetWitnesses();
-                try
+                using (ApplicationEngine engine = ApplicationEngine.Run(script, snapshot, tx, testMode: true))
                 {
-                    tx.CalculateFees();
+                    if (engine.State.HasFlag(VMState.FAULT)) return null;
+                    tx.Gas = Math.Max(engine.GasConsumed - ApplicationEngine.GasFree, 0);
+                    if (tx.Gas > 0)
+                    {
+                        long d = (long)NativeContract.GAS.Factor;
+                        long remainder = tx.Gas % d;
+                        if (remainder > 0)
+                            tx.Gas += d - remainder;
+                        else if (remainder < 0)
+                            tx.Gas -= remainder;
+                    }
                 }
-                catch (InvalidOperationException)
+                UInt160[] hashes = tx.GetScriptHashesForVerifying(snapshot);
+                int size = Transaction.HeaderSize + attributes.GetVarSize() + script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+                foreach (UInt160 hash in hashes)
                 {
-                    continue;
+                    script = GetAccount(hash)?.Contract?.Script ?? snapshot.Contracts.TryGet(hash)?.Script;
+                    if (script is null) continue;
```

Comment:
if we don't find the script, we should return false, or remove the hash

Reply:
In fact this can't happen.

Reply:
Then, i prefer to throw and exception

Reply:
The only case is one put an cosigner in the `Attributes`. In this case, we don't throw exceptions to allow users calculate the fee by themselves.